src/core/server/services/archive/index.ts

```diff
@@ -0,0 +1,328 @@
+import { Collection, Cursor } from "mongodb";
+
+import { MongoContext } from "coral-server/data/context";
+import { StoryNotFoundError } from "coral-server/errors";
+import { Logger } from "coral-server/logger";
+import {
+  createEmptyRelatedCommentCounts,
+  RelatedCommentCounts,
+  updateSharedCommentCounts,
+} from "coral-server/models/comment/counts";
+import { updateSiteCounts } from "coral-server/models/site";
+import { Story } from "coral-server/models/story";
+import {
+  archivedCommentActions,
+  archivedCommentModerationActions,
+  archivedComments,
+  commentActions,
+  commentModerationActions,
+  comments,
+  stories,
+} from "coral-server/services/mongodb/collections";
+import { AugmentedRedis } from "coral-server/services/redis";
+
+const getCommentCounts = (options: {
+  story: Readonly<Story>;
+  negate: boolean;
+}) => {
+  const {
+    story: { commentCounts },
+    negate,
+  } = options;
+  const multiplier = negate ? -1 : 1;
+
+  const result: RelatedCommentCounts = createEmptyRelatedCommentCounts();
+
+  if (commentCounts.action) {
+    for (const key in commentCounts.action) {
+      if (result.action.hasOwnProperty.call(commentCounts.action, key)) {
```

Comment:
```suggestion
      if (Object.prototype.hasOwnProperty.call(commentCounts.action, key)) {
```

For the rest of this file please!